pkg/domain/infra/abi/artifact.go

```diff
@@ -80,17 +81,45 @@ func (ir *ImageEngine) ArtifactPull(ctx context.Context, name string, opts entit
 	return nil, artStore.Pull(ctx, name, *pullOptions)
 }
 
-func (ir *ImageEngine) ArtifactRm(ctx context.Context, name string, _ entities.ArtifactRemoveOptions) (*entities.ArtifactRemoveReport, error) {
+func (ir *ImageEngine) ArtifactRm(ctx context.Context, name string, opts entities.ArtifactRemoveOptions) (*entities.ArtifactRemoveReport, error) {
+	var (
+		namesOrDigests []string
+	)
+	artifactDigests := make([]*digest.Digest, 0)
 	artStore, err := store.NewArtifactStore(getDefaultArtifactStore(ir), ir.Libpod.SystemContext())
 	if err != nil {
 		return nil, err
 	}
-	artifactDigest, err := artStore.Remove(ctx, name)
-	if err != nil {
-		return nil, err
+
+	if opts.All {
+		allArtifacts, err := artStore.List(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, art := range allArtifacts {
+			// Using the digest here instead of name to protect against
+			// an artifact that lacks a name
+			manifestDigest, err := art.GetDigest()
+			if err != nil {
+				return nil, err
+			}
+			namesOrDigests = append(namesOrDigests, manifestDigest.Encoded())
+		}
+	}
+
+	if name != "" {
```

Comment:
Should we detect if `options.All` was also specified here and error?

Reply:
my initial thought was ... we detect this on the client side already which covers local and remote-client use.  So for the RESTFUL side, i would add to the tunnel side?